src/main/java/org/carlmontrobotics/subsystems/Arm.java

```diff
@@ -4,18 +4,79 @@
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.REVLibError;
+import com.revrobotics.SparkAbsoluteEncoder;
+import com.revrobotics.SparkPIDController;
+import com.revrobotics.CANSparkBase;
+import com.revrobotics.CANSparkBase.ControlType;
+import com.revrobotics.SparkPIDController.AccelStrategy;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.math.trajectory.TrapezoidProfile.State;
 import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.XboxController;
+import edu.wpi.first.wpilibj.motorcontrol.MotorController;
 import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import edu.wpi.first.wpilibj2.command.Command;
 import edu.wpi.first.wpilibj2.command.InstantCommand;
 import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 import edu.wpi.first.wpilibj2.command.WaitCommand;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+
+
 
 public class Arm extends SubsystemBase {
+    private final CANSparkMax armMotor1 = MotorControllerFactory.createSparkMax(LEFT_MOTOR_PORT,MotorConfig.NEO);
+    //private final CANSparkMax armMotor2 = MotorControllerFactory.createSparkMax(Constants.Arm.RIGHT_MOTOR_PORT,MotorConfig.NEO);
+    //there is only one arm motor
+    //test
+
+    private final SimpleMotorFeedforward armFeed = new SimpleMotorFeedforward(kS, kV);
+    private final SparkAbsoluteEncoder armEncoder = armMotor1.getAbsoluteEncoder(SparkAbsoluteEncoder.Type.kDutyCycle);
+    private final SparkPIDController armPID = armMotor1.getPIDController();
+    public static TrapezoidProfile.State[] goalState = { 
+      new TrapezoidProfile.State(Constants.Arm.intakeAngle, 0), 
+      new TrapezoidProfile.State(Constants.Arm.ampAngle, 0),  
+      new TrapezoidProfile.State(Constants.Arm.placeholderSpeakerAngle1, 0),
+      new TrapezoidProfile.State(Constants.Arm.placeholderSpeakerAngle2, 0),
+      new TrapezoidProfile.State(Constants.Arm.placeholderSpeakerAngle3, 0),
+      new TrapezoidProfile.State(Constants.Arm.climberUpAngle, 0),
+      new TrapezoidProfile.State(Constants.Arm.climberDownAngle, 0),
+    };
+    
+    TrapezoidProfile profile = new TrapezoidProfile(Constants.Arm.trapConstraints);
+    
     public Arm() {
+      armPID.setP(Constants.Arm.kP);
+      armPID.setI(Constants.Arm.kI);
+      armPID.setD(Constants.Arm.kD);
+      armPID.setFF(kV,0);
+      armPID.setIZone(IZONE); //<-- zero for now
+      SmartDashboard.putNumber("kp", kP);
+      SmartDashboard.putNumber("kd", kD);
+      SmartDashboard.putNumber("ki", kI);
+      SmartDashboard.putNumber("Shooter RPS", 0);
```

Comment:
we never get these so there's no point to put them.
add gets in periodic if we want to keep these.

Reply:
done